contracts/wrapperContracts/WrapFeeBurner.sol

```diff
@@ -0,0 +1,227 @@
+        require(minRate > 0);
+
+        //update data tracking
+        setNewData(KNC_RATE_RANGE_INDEX);
+
+        kncPerEth.pendingMinRate = minRate;
+        kncPerEth.pendingMaxRate = maxRate;
+    }
+
+    function approveKNCRateRange(uint nonce) public onlyOperator {
+        if (addSignature(KNC_RATE_RANGE_INDEX, nonce, msg.sender)) {
+            // can perform operation.
+            kncPerEth.minRate = kncPerEth.pendingMinRate;
+            kncPerEth.maxRate = kncPerEth.pendingMaxRate;
+        }
+    }
+
+    function getPendingKNCRateRange() public view returns(uint minRate, uint maxRate, uint nonce) {
+        address[] memory signatures;
+        minRate = kncPerEth.pendingMinRate;
+        maxRate = kncPerEth.pendingMaxRate;
+        (signatures, nonce) = getDataTrackingParameters(KNC_RATE_RANGE_INDEX);
+
+        return(minRate, maxRate, nonce);
+    }
+
+    function getKNCRateRange() public view returns(uint minRate, uint maxRate) {
+        minRate = kncPerEth.minRate;
+        maxRate = kncPerEth.maxRate;
+        return(minRate, maxRate);
+    }
+
+    function setKNCPerEthRate(uint kncPerEther) public onlyOperator {
+        require(kncPerEther >= kncPerEth.minRate);
+        require(kncPerEther <= kncPerEth.maxRate);
+        feeBurnerContract.setKNCRate(kncPerEther);
+    }
+
+    function getKNCRateRangeSignatures() public view returns (address[] signatures) {
+        uint nonce;
+        (signatures, nonce) = getDataTrackingParameters(KNC_RATE_RANGE_INDEX);
+        return(signatures);
+    }
+
+    //set reserve data
+    //////////////////
+    function setPendingReserveData(address _reserve, uint feeBps, address kncWallet) public onlyOperator {
+        require(_reserve != address(0));
+        require(kncWallet != address(0));
+        require(feeBps > 0);
+
+        addReserve.reserve = _reserve;
+        addReserve.feeBps = feeBps;
+        addReserve.kncWallet = kncWallet;
+        setNewData(ADD_RESERVE_INDEX);
+    }
+    
+    function approveAddReserveData(uint nonce) public onlyOperator {
+        if (addSignature(ADD_RESERVE_INDEX, nonce, msg.sender)) {
+            // can perform operation.
+            feeBurnerContract.setReserveData(addReserve.reserve, addReserve.feeBps, addReserve.kncWallet);
+        }
+    }
+
+    function getPendingAddReserveData() public view
+        returns(address _reserve, uint feeBps, address kncWallet, uint nonce)
+    {
+        address[] memory signatures;
+        (signatures, nonce) = getDataTrackingParameters(ADD_RESERVE_INDEX);
+        return(addReserve.reserve, addReserve.feeBps, addReserve.kncWallet, nonce);
+    }
+
+    function getAddReserveSignatures() public view returns (address[] signatures) {
+        uint nonce;
+        (signatures, nonce) = getDataTrackingParameters(ADD_RESERVE_INDEX);
+        return(signatures);
+    }
+
+    //wallet fee
+    /////////////
+    function setPendingWalletFee(address wallet, uint feeInBps) public onlyOperator {
```

Comment:
consider renaming feeInBps  to feeBps, didn't see the "In" style used elsewhere in this contract.